components/object_store/src/disk_cache.rs

```diff
@@ -145,138 +216,214 @@ impl DiskCache {
 
         Ok(Self {
             root_dir,
-            cache: PartitionedMutexAsync::try_new(init_lru, partition_bits, SeaHasherBuilder {})?,
+            meta_cache: PartitionedMutex::try_new(init_lru, partition_bits, SeaHasherBuilder {})?,
         })
     }
 
-    /// Update the cache.
-    ///
-    /// The returned value denotes whether succeed.
-    // TODO: We now hold lock when doing IO, possible to release it?
-    async fn update_cache(&self, key: String, value: Option<Bytes>) -> bool {
-        let mut cache = self.cache.lock(&key).await;
+    fn insert_page_meta(&self, filename: String, page_meta: PageMeta) -> Option<String> {
+        let mut cache = self.meta_cache.lock(&filename);
         debug!(
-            "Disk cache update, key:{}, len:{}, cap_per_part:{}.",
-            &key,
+            "Update the meta cache, file:{filename}, len:{}, cap_per_part:{}",
             cache.cap(),
-            cache.len(),
+            cache.len()
         );
 
-        // TODO: remove a batch of files to avoid IO during the following update cache.
-        if cache.len() >= cache.cap() {
-            let (filename, _) = cache.pop_lru().unwrap();
-            let file_path = std::path::Path::new(&self.root_dir)
-                .join(filename)
+        cache
+            .push(filename, page_meta)
+            .map(|(filename, _)| filename)
+    }
+
+    async fn insert_data(&self, filename: String, value: Bytes) {
+        let page_meta = {
+            let file_size = PageFileEncoder::encoded_size(value.len());
+            PageMeta { file_size }
+        };
+        let evicted_file = self.insert_page_meta(filename.clone(), page_meta);
+
+        let do_persist = || async {
+            if let Err(e) = self.persist_bytes(&filename, value).await {
+                error!("Failed to persist cache, file:{filename}, err:{e}");
+            }
+        };
+
+        if let Some(evicted_file) = evicted_file {
+            if evicted_file == filename {
```

Comment:
LRU cache will not evict the key as you just insert, if you want avoid persist, you can use `cache.contains`.

Reply:
```
    /// Pushes a key-value pair into the cache. If an entry with key `k` already exists in
    /// the cache or another cache entry is removed (due to the lru's capacity),
    /// then it returns the old entry's key-value pair. Otherwise, returns `None`.
```